redfish-core/lib/bios.hpp

```diff
@@ -53,6 +260,476 @@ inline void requestRoutesBiosService(App& app)
             std::bind_front(handleBiosServiceGet, std::ref(app)));
 }
 
+/**
+ * BiosSettings class supports handle GET/PATCH method for
+ * BIOS configuration pending settings.
+ */
+inline void requestRoutesBiosSettings(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/<str>/Bios/Settings")
+        .privileges({{"ConfigureManager"}})
+        .methods(boost::beast::http::verb::get)(
+            [&app](const crow::Request& req,
+                   const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                   const std::string& systemName) {
+        if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+        {
+            return;
+        }
+
+        if (systemName != "system")
+        {
+            messages::resourceNotFound(asyncResp->res, "ComputerSystem",
+                                       systemName);
+            return;
+        }
+        asyncResp->res.jsonValue["@odata.id"] =
+            "/redfish/v1/Systems/system/Bios/Settings";
+        asyncResp->res.jsonValue["@odata.type"] = "#Bios.v1_1_0.Bios";
+        asyncResp->res.jsonValue["Name"] = "Bios Settings";
+        asyncResp->res.jsonValue["Id"] = "BiosSettings";
+        asyncResp->res.jsonValue["AttributeRegistry"] = "BiosAttributeRegistry";
+        nlohmann::json attributes(nlohmann::json::value_t::object);
+        asyncResp->res.jsonValue["Attributes"] = attributes;
+
+        crow::connections::systemBus->async_method_call(
+            [asyncResp](const boost::system::error_code ec,
+                        const dbus::utility::MapperGetObject& getObjectType) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("ObjectMapper::GetObject call failed: {}", ec);
+                messages::internalError(asyncResp->res);
+
+                return;
+            }
+            std::string service = getObjectType.begin()->first;
+
+            crow::connections::systemBus->async_method_call(
+                [asyncResp](const boost::system::error_code ec1,
+                            const std::variant<PendingAttributesType>&
+                                retPendingAttributes) {
+                if (ec1)
+                {
+                    BMCWEB_LOG_ERROR("getBiosSettings DBUS error: {}", ec1);
```

Comment:
this should be a warning

Reply:
if it is an internal error, ERROR is correct here

Reply:
@cmjishnu modify here to use Warning

Reply:
Done